file_example.c

```diff
@@ -97,8 +97,8 @@ file_handler_run(void *arg)
 
 		/* process command */
 		result = file_handle_cmd(h->dev, cmd);
-		pthread_mutex_lock(&state->completion_mtx);
 		tcmulib_command_complete(h->dev, cmd, result);
+		pthread_mutex_lock(&state->completion_mtx);
 		tcmulib_processing_complete(h->dev);
 		pthread_mutex_unlock(&state->completion_mtx);
```

Comment:
Thanks for fixing and debugging bgly.

Was the locking call change in the completion paths here and below related to the bug, or just a change because  libtcmu's tcmulib_command_complete locks for the handler now?